operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/AbstractDependentResource.java

```diff
@@ -0,0 +1,33 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+public abstract class AbstractDependentResource<R, P extends HasMetadata>
+    implements DependentResource<R, P> {
+
+  @Override
+  public void reconcile(P primary, Context context) {
+    var actual = getResource(primary);
+    var desired = desired(primary, context);
+    if (actual.isEmpty()) {
```

Comment:
How do you specify that a dependent should never be created with this? It's a pretty common use case that you only want to update a dependent but not create it and I don't see how you can express this with this implementation.

Reply:
That would be probably a feature flag, we can add to this default implementation, and be part of the logic. But we can also extend this or the KubernetesDepenendtResource and override the reconcile method. This is how is exactly meant, like the reconcile logic could be adjusted to special cases.

Reply:
A common use case should be supported by default without people having to do extra things to support it. I'd argue that the default should indeed be the case where we don't want to create a dependent, just update it or query its information. Implementing something is needed when creating resources so it's normal that it requires input. I think that the previous implementation was cleaner on that part.

Reply:
Actually, the good part is that we can model that separatelly with other subclass maybe of the DependentResource interface. So it's more explicit. Will take a look and see how it looks.

Reply:
I would really prefer we avoid forcing people to learn about a class hierarchy to use dependent resources.

Reply:
So currently there is just `KubernetesDependentResource` and an abstract above with generic logic. And one for standalone.  

Will take a look if the Kubernetes and the StandaloneOne could be unified. Probably yes.